### DISCUSSION TITLE
Curious about pvc clone

### DISCUSSION THREAD

**Discussion by qkfrksvl** [ceph/ceph-csi#4940] [Q&A]
If you look at this document, snaphost and clone are used twice. It seems like replication is completed just once, so why is it doing it twice?

https://github.com/ceph/ceph-csi/blob/devel/docs/snap-clone.md#clone-rbd-pvc

**Comment by Rakshith-R** [comment]
> If you look at this document, snaphost and clone are used twice. It seems like replication is completed just once, so why is it doing it twice?
> 
> https://github.com/ceph/ceph-csi/blob/devel/docs/snap-clone.md#clone-rbd-pvc

Flatten operation is affected by IO and there are limits to number of snapshots and parent->child->grand-child rbd image chain.

The intermediate temp image exists to be flattened in case the image chain(clone depth) grows too long or there are too many snapshots on the parent image.
